catalogue/webapp/components/IIIFSearchWithin/IIIFSearchWithin.tsx

```diff
@@ -177,49 +170,51 @@ const IIIFSearchWithin: FunctionComponent<Props> = ({ mainViewerRef }) => {
               .filter(resource => resource?.resource?.chars);
             // Get the index of the canvas the hits appear on
-            const index = canvases.findIndex(canvas => {
+            const index = canvases?.findIndex(canvas => {
               const matchingPathname = matchingResources?.[0]?.on || '';
               return (
                 new URL(matchingPathname).pathname ===
                 new URL(canvas.id).pathname
               );
             });
-            const matchingCanvas = (index && canvases[index]) || undefined;
+            const matchingCanvas = (index && canvases?.[index]) || undefined;
             return (
               <ListItem key={i}>
-                <SearchResult
-                  onClick={() => {
-                    if (index) {
-                      setIsMobileSidebarActive(false);
-                      setActiveIndex(index || 0);
-                      mainViewerRef &&
-                        mainViewerRef.current &&
-                        mainViewerRef.current.scrollToItem(index || 0, 'start');
-                    }
-                  }}
-                >
-                  <HitData v={{ size: 's', properties: ['margin-bottom'] }}>
-                    {`${
-                      index &&
-                      `Found on image ${index + 1} / ${
-                        canvases && canvases.length
-                      }`
-                    } ${
-                      matchingCanvas?.label?.trim() !== '-'
-                        ? ` (page ${matchingCanvas?.label})`
-                        : ''
-                    }`}
-                  </HitData>
-                  <span role="presentation">…{hit.before}</span>
-                  <span
-                    style={{
-                      background: '#944aa0',
-                      color: 'white',
-                    }}
+                <SearchResult>
+                  <NextLink
+                    {...itemLink(
+                      {
+                        workId: work.id,
+                        manifest: query.manifestParam,
+                        canvas: arrayIndexToQueryParam(index || 0),
+                      },
+                      'search_within_result'
+                    )}
+                    onClick={() => setIsMobileSidebarActive(false)}
                   >
-                    {hit.match}
-                  </span>
-                  <span role="presentation">{hit.after}...</span>
+                    <HitData v={{ size: 's', properties: ['margin-bottom'] }}>
+                      {`${
+                        index &&
+                        `Found on image ${arrayIndexToQueryParam(index)} / ${
+                          canvases && canvases.length
+                        }`
+                      } ${
+                        matchingCanvas?.label?.trim() !== '-'
+                          ? ` (page ${matchingCanvas?.label})`
+                          : ''
+                      }`}
```

Comment:
I find this a little hard to grasp/read, could it be pulled into a smaller component or the text rendered from a function?